### PR TITLE
chore: add variable.less to support css variable

### PR DESCRIPTION
**Pull request by simplejason**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #7389

## What is the new behavior?
- add `@ctrl/tinycolor`
- support setting css variable via `NzConfigService`
- sync codes about `css variable` from `ant-design`

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by hullis — APPROVED** [review]
LGTM. Glad to see it get landed!

File: docs/customize-theme-variable.zh-CN.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+```
+
+注：如果你使用了 `babel-plugin-import`，需要将其去除。
+
+### 静态方法配置
+
+调用 ConfigProvider 配置方法设置主题色：
+
+```ts
+import { ConfigProvider } from 'antd';
```
Comment (hullis) [inline docs/customize-theme-variable.zh-CN.md:31]: antd

**Review by hullis — CHANGES_REQUESTED** [review]
Docs are for the React version.

File: docs/customize-theme-variable.zh-CN.md (hunk 1 above)
Comment (simplejason) [reply docs/customize-theme-variable.zh-CN.md]: 已经支持 nzConfigService，还有些测试我补齐一下

**Review by yangjunhan — APPROVED** [review]
LGTM

**Pull request merged**